File: logistica.m
% Ian Mu;oz Nu;ez - Funcion sigmoidal

close all
clear
clc

n = 1000;

v = linspace(-10, 10, n);

a = 0.2;

phi = zeros(5, n);
for i=1:5
    for j=1:n
        phi(i, j) = 1/( 1 + exp(-a*v(j)) );
    end
    a = a + 0.2;
end

figure(1)
hold on
grid on

plot(v, phi(1, :), 'r', 'LineWidth', 2)
plot(v, phi(2, :), 'b', 'LineWidth', 2)
plot(v, phi(3, :), 'g', 'LineWidth', 2)
plot(v, phi(4, :), 'y', 'LineWidth', 2)
plot(v, phi(5, :), 'm', 'LineWidth', 2)

title("Funcion sigmoidal", 'FontSize', 20)
xlabel('v', 'FontSize', 15)
ylabel('\phi(v)', 'FontSize', 15)
legend('0.2', '0.4', '0.6', '0.8', '1')

File: logistica.py
# Ian Mu;oz Nu;ez - Funcion logistica

import matplotlib.pyplot as plt
import numpy as np

n = 1000

v = np.linspace(-10, 10, n)

a = 0.2

phi = np.zeros((5, n))
for i in range(0, 5):
    for j in range(0, n):
        phi[i, j] = 1 / ( 1 + np.exp(-a * v[j]) )
    a += 0.2

plt.figure(1)
plt.grid()

plt.plot(v, phi[0, :], 'r', linewidth=2, label=r'0.2')
plt.plot(v, phi[1, :], 'b', linewidth=2, label=r'0.4')
plt.plot(v, phi[2, :], 'g', linewidth=2, label=r'0.6')
plt.plot(v, phi[3, :], 'y', linewidth=2, label=r'0.8')
plt.plot(v, phi[4, :], 'm', linewidth=2, label=r'1')

plt.title("Funcion logistica", fontsize=20)
plt.xlabel('v', fontsize=15)
plt.ylabel(r'$\phi$(v)', fontsize=15)

plt.legend()
plt.show()

